src/librustc_span/symbol.rs

```diff
@@ -102,9 +102,13 @@ symbols! {
         Union:              "union",
     }
 
-    // Symbols that can be referred to with rustc_span::sym::*. The symbol is
-    // the stringified identifier unless otherwise specified (e.g.
-    // `proc_dash_macro` represents "proc-macro").
+    // Pre-interned symbols that can be referred to with `rustc_span::sym::*`.
+    //
+    // The symbol is the stringified identifier unless otherwise specified, in
+    // which case the name should mention the non-identifier punctuation.
+    // E.g. `sym::proc_dash_macro` represents "proc-macro", and it shouldn't be
+    // called `sym::proc_macro` because then it's easy to mistakenly think it
+    // represents "proc_macro".
```

Comment:
Perhaps Zoxc was right about [introducing a separate namespace for these cases](https://github.com/rust-lang/rust/pull/60630#discussion_r282761148) to avoid confusion. Not sure.

Reply:
ooh, that's a good idea!

Reply:
Sure, but best for a follow-up.